src/app/components/category-products/category-products.component.ts

```diff
@@ -1,11 +1,105 @@
-import { ChangeDetectionStrategy, Component, ViewEncapsulation } from '@angular/core';
+import {
+  ChangeDetectionStrategy,
+  Component,
+  ViewEncapsulation,
+} from '@angular/core';
+import { FormControl } from '@angular/forms';
+import { ActivatedRoute } from '@angular/router';
+import { Observable, combineLatest, of } from 'rxjs';
+import {  map, shareReplay, startWith, switchMap } from 'rxjs/operators';
+import { CategoryModel } from '../../models/category.model';
+import { ProductModel } from '../../models/product.model';
+import { CategoriesService } from '../../services/categories.service';
+import { ProductsService } from '../../services/products.service';
 
 @Component({
   selector: 'app-category-products',
   styleUrls: ['./category-products.component.scss'],
   templateUrl: './category-products.component.html',
   encapsulation: ViewEncapsulation.Emulated,
-  changeDetection: ChangeDetectionStrategy.OnPush
+  changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class CategoryProductsComponent {
+  readonly selectedSortingOption: FormControl = new FormControl(
+    'Featured'
+  );
+  readonly sortingOptions$: Observable<string[]> = of([
+    'Featured',
+    'Price: Low To High',
+    'Price: High To Low',
+    'Avg. Rating',
+  ]);
+
+  readonly categoryData$: Observable<CategoryModel> =
+    this._activatedRoute.params.pipe(
+      switchMap((data) => this._categoriesService.getOne(data['categoryId'])),
+      shareReplay(1)
+    );
+  readonly categories$: Observable<CategoryModel[]> =
+    this._categoriesService.getAll();
+
+  readonly productsInCategory$: Observable<ProductModel[]> = combineLatest([
+    this._productsService.getAll(),
+    this.categoryData$,
+    this.selectedSortingOption.valueChanges.pipe(
+      startWith('Featured')
+    ),
+  ]).pipe(
+    map(
+      ([products, category, sortingOption]: [
+        ProductModel[],
+        CategoryModel,
+        string
+      ]) => {
+        if (sortingOption === 'Price: Low To High') {
```

Comment:
this is quite complex (many ifs) and error prone (compare 2 strings). I think you  can do it better here :)